Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -39,8 +39,92 @@ private void BuildScrollArea(ImGui gui) {
+            parents.Add(t);
+        }
+        if (o is ProductionTable page) {
+            Dictionary<ProductionLink, List<(RecipeRow row, float flow)>> childLinks = [], parentLinks = [], otherLinks = [];
+            List<(RecipeRow row, float flow)> unlinked = [], table;
+            foreach (var (r, relationLinks) in
+                GetAllRecipes(page, link.owner)
+                .Select(e => (e, IsParrent(e, parents) ? parentLinks : otherLinks))
+                .Concat(GetAllRecipes(link.owner).Select(e => (e, childLinks)))) {
+                if (link.capturedRecipes.Any(e => e == r)
+                    || (!r.Ingredients.Any(e => e.Goods == link.goods)
+                    && !r.Products.Any(e => e.Goods == link.goods)
+                    && !(r.fuel is not null && r.fuel == link.goods))) {
+                    continue;
+                }
+                float localFlow = DetermineFlow(link.goods, r);
+
+                if ((r.FindLink(link.goods, out var otherLink) && otherLink != link)) {
+                    if (!relationLinks.ContainsKey(otherLink)) {
+                        relationLinks.Add(otherLink, []);
+                    }
+                    table = relationLinks[otherLink];
+                }
+                else {
+                    table = unlinked;
+                }
+
+                if (localFlow > 0) {
+                    table.Add((r, localFlow));
+                }
+                else if (localFlow < 0) {
+                    table.Add((r, localFlow));
+                }
+                else {
+                    table.Add((r, 0));
+                }
+            }
+            var color = 1;
+            if (childLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Child links: ", Font.subheader);
+                foreach (var relTable in childLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (parentLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Parent links: ", Font.subheader);
+                foreach (var relTable in parentLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (otherLinks.Values.Any(e => e.Any())) {
+                gui.BuildText("Unrelated links: ", Font.subheader);
+                foreach (var relTable in otherLinks.Values) {
+                    BuildFlow(gui, relTable, relTable.Sum(e => Math.Abs(e.flow)), false, color++);
+                }
+            }
+            if (unlinked.Any()) {
+                gui.BuildText("Unlinked: ", Font.subheader);
+                BuildFlow(gui, unlinked, 0, false);
+            }
+        }
+    }
+
+    private bool IsParrent(RecipeRow row, List<ModelObject> parents) => row.Ingredients.Select(e => e.Link).Concat(row.Products.Select(e => e.Link)).Append(row.FuelInformation.Link)
```

Comment:
Should this function be part of `RecipeRow`, so you can do `recipeRow->IsParent(parents)` instead?

Reply:
Mmm, this is a quite specific `IsParent` function for finding parents of a link. It scans all ingredients, products and fuel and checks them for the `link` owner (the `link` isn't even a parameter, it's a local field).
I've renamed it to `isLinkParent` instead.